### ISSUE TITLE
[Feature]: Ability to retrieve encoded Identity Token from OIDC authenticated user

### ISSUE THREAD

**Issue by Peterkrol12** [appsmithorg/appsmith#25293]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

The ability to access the encoded JWT identity token from an OIDC authenticated user. This means that instead of decoding the received identity token into a 'human readable' object, just straight up returning the encoded JWT. Perhaps this can also be hidden under a placeholder, the same way the access token is hidden.

### Why should this be worked on?

Currently, one can only retrieve the decoded identity token of a OIDC authenticated user. While this is a nice to have and certainly improves some things in the workflow, it also brings limitations.

Our use case requires the generation of [AWS V4 signature headers](https://docs.aws.amazon.com/AmazonS3/latest/API/sig-v4-authenticating-requests.html), which requires AWS credentials to create the signature. Currently, there is no built-in support for AWS V4 headers, although there have been some comments about it in [this issue](https://github.com/appsmithorg/appsmith/issues/16424). This means that the generation of the headers is to be done by using the AWS JavaScript SDK.

Our OIDC flow uses AWS Cognito to authenticate users. The returned identity token includes some crucial information which is needed to retrieve a cognito 'identity id' using [this call](https://docs.aws.amazon.com/cognitoidentity/latest/APIReference/API_GetId.html). The cognito identity id can then be used to retrieve AWS credentials using [this call](https://docs.aws.amazon.com/cognitoidentity/latest/APIReference/API_GetCredentialsForIdentity.html).

In this flow, the encoded JWT identity token is needed for the cognito identity id request to succeed. I have tried to request the cognito identity id with the decoded JWT identity token, which returned an error related to the formatting of the identity token.

**Comment by infinitetrooper** [comment]
@Peterkrol12  - I've added this to our immediate roadmap but, we need a few weeks to do solution exploration on this before we can confirm support for it. I'll keep this ticket updated with the status of our explorations.

**Comment by thinknirmal** [comment]
This is crucial for us as well. We are using Auth0 for authentication, but we need the JWT to be passed as Bearer token header to the GraphQL API. Currently we are stuck with no way of doing it.

**Comment by infinitetrooper** [comment]
@thinknirmal  - We're actively exploring a solution for this right now. We'll post updates on the progress here.

**Comment by Peterkrol12** [comment]
As this is now marked as 'closed', is this issue resolved and is a solution implemented?

Could we receive some information regarding the solution?

Thanks a lot for your time and quick remediation :)

**Comment by infinitetrooper** [comment]
@Peterkrol12  - This is currently under QA and should be out in a week or two. You should be able to access the encoded token under `{{appsmith.use.rawIdToken}}` when it's available. We'll update our documentation as well closer to release.

**Comment by Peterkrol12** [comment]
@infinitetrooper - That is great to hear, thanks for the reply. We'll be sure to immediately check it out and implement it!
